library/beaker_provisioner.py

```diff
@@ -0,0 +1,419 @@
+        })
+
+        resp = self.session.post(url, data=params, headers=headers)
+
+        if is_fail(resp.status_code, 201):
+            return True
+
+        if wait_for_host and self._provision_waiter(
+                last_pre_provison_activity, provision_timeout):
+            return True
+
+        return False
+
+    def _get_last_system_activity(self):
+        """
+        Return the ID of the last system's activity
+        """
+
+        headers = {'accept': 'application/json'}
+        url = self.base_url + '/activity/system'
+        self.session.params = {'q': 'system:' + self.fqdn}
+        resp = self.session.get(url, headers=headers)
+
+        return json.loads(resp.text)['entries'][0]
+
+    def _provision_waiter(self, pre_provision_activity_id, timeout):
+        """
+        Waiting until the provisioning process id done and the system responses
+        tp ping requests
+
+        :param pre_provision_activity_id: Last system's activity ID
+        :param timeout: Max waiting time for process to be done
+        :return:
+        """
+
+        start_time = time.time()
+
+        # Provision waiter
+        while (time.time() - start_time) < timeout:
+            time.sleep(WAIT_BETWEEN_PROVISION_CHECKS)
+            last_activity = self._get_last_system_activity()
+            if (last_activity['id'] != pre_provision_activity_id['id']) \
+                    and last_activity['action'] == 'clear_netboot':
+                break
+        else:
+            return True
+
+        # Ping waiter
+        while (time.time() - start_time) < timeout:
+            command = "ping -q -c 30 -w 300 {0}".format(self.fqdn)
+            if not subprocess.call(command.split(" ")):
+                return False
+        else:
+            return True
+
+    def _build_url(self, action, fqdn=None):
+        """
+        Build URL by concatenating the base URL of the beaker server with
+        the specific action string
+
+        :param action: An action from the class variable ACTIONS
+        :param fqdn: The system's fully qualified domain name
+        :return: String of the absolute URL
+        """
+
+        url = self.REQ_URL.get(action)
+        if fqdn:
+            url = urljoin(SYS_FQDN_PATH.format(fqdn=fqdn), url)
+
+        return urljoin(self.base_url, url)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            server_url=dict(required=True),
+            username=dict(default='admin'),
+            password=dict(required=True),
+            fqdn=dict(required=True),
+            action=dict(required=True, choices=['provision', 'release']),
+            distro_tree_id=dict(default=71576)))
+
+    beaker_client = BeakerManager(url=module.params['server_url'],
+                                  username=module.params['username'],
+                                  password=module.params['password'],
+                                  fqdn=module.params['fqdn'])
```

Comment:
When creating a class called "manager" you expect this class to provide a service of some sort, if you init it with a FQDN, that would be a BeakerMachine class.. not a BeakerManager